### ISSUE TITLE
Needs to support multiple targets

### ISSUE THREAD

**Issue by luckman212** [dennypage/dpinger#22]
Hi Denny

Could you possibly review https://redmine.pfsense.org/issues/4354 and let us know your thoughts? I am more than happy to donate financially to the cause if that helps. I had a bad outage yesterday on a pfSense 2.3.2_p1 system that was due to a false positive on dpinger. This caused a gateway to get marked "down" when it wasn't really an ISP issue. We believe it was just an 8.8.8.8 problem.

It has been suggested by myself and others that allowing additional targets would really help reduce these edge cases and improve reliability.

**Comment by dennypage** [comment]
Hey Luke,

I understand what you are trying to accomplish, but dpinger is not the place to implement that type of logic. Dpinger is a latency/loss monitor, not a decision maker. The concept of monitoring multiple targets, and deciding what if to do if a value for one or both of them reaches a certain threshold, belongs in a higher level. A good example of this higher level decision making is the already exiting latency/loss thresholds present in pfSense 2.3. Another example, and perhaps more direct to the point, is the action disable option offered in 2.4.

There are other less important issues, such as how do you report the combined results? Do you average them? Anything you do in terms of averaging hides information from the upper levels which in general is not a good idea. For example, knowing that the average latency between the two links jumped from 18ms to 80ms isn't sufficient. You really need to know which link is the source of the problem. There are other issues: If one of the sending interfaces is down, do you report an error or do you ignore it?; If one link has high latency and the other link has high loss, what do you report? 

In short, the concept of monitoring multiple targets, and combining their results in a decision making process to determine whether a gateway is "down" is something that has to be implemented in pfSense itself. I'll be happy to look at this in pfSense as time permits, although I believe that Phil has already said that he was going to explore it.

Best,
Denny
